### PR TITLE
c++20 concept: enforcing type for get_entropy implementers.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

**Comment by devnexen** [comment]
Note I did not trigger the formatting seems the whole file needs to.

File: src/pal/pal_noalloc.h [hunk 1]
```diff
@@ -81,6 +81,11 @@ namespace snmalloc
     {
       memset(p, 0, size);
     }
+
+    static uint64_t get_entropy64()
```
Comment (nwf) [inline src/pal/pal_noalloc.h:85]: This hunk becomes unnecessary with the above.

File: src/pal/pal_concept.h [hunk 2]
```diff
@@ -91,6 +97,7 @@ namespace snmalloc
     ConceptPAL_static_members<PAL> &&
     ConceptPAL_error<PAL> &&
     ConceptPAL_memops<PAL> &&
+    ConceptPAL_get_entropy64<PAL> &&
```
Comment (nwf) [inline src/pal/pal_concept.h:100]: Please gate this on the `Entropy` `PalFeatures` flag:

```suggestion
    (!pal_supports<Entropy, PAL> ||
      ConceptPAL_get_entropy64<PAL>) &&
```

**Review by nwf — CHANGES_REQUESTED** [review]

**Review by nwf — APPROVED** [review]
LGTM

File: src/pal/pal_concept.h [hunk 3]
```diff
@@ -91,6 +97,8 @@ namespace snmalloc
     ConceptPAL_static_members<PAL> &&
     ConceptPAL_error<PAL> &&
     ConceptPAL_memops<PAL> &&
+    (!pal_support<Entropy>, PAL> ||
```
Comment (nwf) [inline src/pal/pal_concept.h:100]: Just a small typo:

```suggestion
    (!pal_supports<Entropy>, PAL> ||
```

**Review by nwf — CHANGES_REQUESTED** [review]

File: src/pal/pal_concept.h (hunk 3 above)
Comment (devnexen) [reply src/pal/pal_concept.h]: yes forgot to retest c++20 build :)

**Review by nwf — APPROVED** [review]

**Pull request merged**